extractor/extractor.cpp

```diff
@@ -253,6 +262,57 @@ int extractor::run()
 
         TIMER_STOP(extracting);
         SimpleLogger().Write() << "extraction finished after " << TIMER_SEC(extracting) << "s";
+    }
+    catch (std::exception &e)
```

Comment:
Always catch by const-ref. All the member functions you want to invoke (you can invoke) are marked const.